src/kompos/hierarchical/composition_config_generator.py

```diff
@@ -163,54 +164,74 @@ def get_sh_command(
         return command
 
 
-class PreConfigGenerator(HierarchicalConfigGenerator, object):
+class PreConfigGenerator(HierarchicalConfigGenerator):
 
     def __init__(self, excluded_config_keys, filtered_output_keys):
         super(PreConfigGenerator, self).__init__()
         self.excluded_config_keys = excluded_config_keys
         self.filtered_output_keys = filtered_output_keys
 
     def pre_generate_config(self, config_path, composition):
-        return self.generate_config(config_path=get_config_path(config_path, composition),
-                                    exclude_keys=self.excluded_config_keys,
-                                    filters=self.filtered_output_keys)
+        return self.generate_config(
+            config_path=get_config_path(config_path, composition),
+            exclude_keys=self.excluded_config_keys,
+            filters=self.filtered_output_keys,
+            skip_interpolation_validation=True,
+            skip_secrets=True
+        )
 
 
-class TerraformConfigGenerator(HierarchicalConfigGenerator, object):
+class TerraformConfigGenerator(HierarchicalConfigGenerator):
 
     def __init__(self, excluded_config_keys, filtered_output_keys):
         super(TerraformConfigGenerator, self).__init__()
         self.excluded_config_keys = excluded_config_keys
         self.filtered_output_keys = filtered_output_keys
 
-    def generate_files(self, config_path, composition_path, composition):
+    def generate_files(self, himl_args, config_path, composition_path, composition):
         config_path = get_config_path(config_path, composition)
         composition_path = get_composition_path(composition_path, composition)
-        self.generate_provider_config(config_path, composition_path)
-        self.generate_variables_config(config_path, composition_path)
 
-    def generate_provider_config(self, config_path, composition_path):
+        self.generate_provider_config(himl_args, config_path, composition_path)
+        self.generate_variables_config(himl_args, config_path, composition_path)
+
+    def generate_provider_config(self, himl_args, config_path, composition_path):
         output_file = os.path.join(composition_path, TERRAFORM_PROVIDER_FILENAME)
         logger.info('Generating terraform config %s', output_file)
 
         filters = self.filtered_output_keys.copy() + ["provider", "terraform"]
+        try:
+            excluded = self.excluded_config_keys.copy() + himl_args.exclude
+        except:
```

Comment:
What exception should be caught here?

Reply:
himl_args only exists if you specify the --himl flag for terraform runner, I'm not happy with how this looks

Reply:
Yes, I think it would be better to catch a specific exception instead of a catch-all. It help with readability also.